### PR TITLE
Add signup and form components

### PR DESCRIPTION
**Pull request by Ormabe**

### CODE REVIEW

File: frontend/components/main/signup.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            label="Google Link"
+            style={styles.button}
+            icon={<FaGoogle />}
+          />
+
+          </Dialog>
+        </div>
+      </MuiThemeProvider>
+    )
+
```
Comment (nmadd) [inline frontend/components/main/signup.js:119]: Overall these components look good, but there's a lot of repetitive code. How can we DRY this out?

For example, we repeat the 'RaisedButton' component four times, and all we change each team are the properties. How can we DRY that out and not write so much repetitive code?

File: frontend/components/main/signupform.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    const value = target.value
+    const name = target.name
+
+    this.setState({ [name]: value })
+	},
+  enterUser(e) {
+  e.preventDefault()
+  const info = this.state
+  const that = this
+
```
Comment (nmadd) [inline frontend/components/main/signupform.js:40]: clean up the above formatting/spacing

File: frontend/components/main/signupform.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        <FlatButton
+          label="Submit"
+          type="submit"
+          backgroundColor={'#AA0000'}
+          color={'#FFFFFF'}
+          style={{color: '#FFFFFF'}}
+        />
+      </form>
+    </MuiThemeProvider>
+    )
```
Comment (nmadd) [inline frontend/components/main/signupform.js:92]: DRY this componenet out as well

File: frontend/components/main/signup.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        opacity: 0,
+      },
+      floatingLabelStyle: {
+        color: "#444444",
+      },
+      floatingLabelFocusStyle: {
+        color: "#000000",
+      }
+    };
+
```
Comment (nmadd) [inline frontend/components/main/signup.js:71]: store styles in a separate file and import them in

File: frontend/components/main/signup.js (hunk 1 above)
Comment (Ormabe) [reply frontend/components/main/signup.js:119]: I am using Material-UI for the forms, and when I tried to replace the repetitive field with "input" it broke the code. I will move the styling to a separate sheet

File: frontend/components/main/signupform.js (hunk 3 above)
Comment (Ormabe) [reply frontend/components/main/signupform.js:92]: Please see the answer for the signup - same scenario applies 😄

File: frontend/components/main/signupform.js (hunk 2 above)
Comment (Ormabe) [reply frontend/components/main/signupform.js:40]: Cleaned

File: frontend/components/main/signup.js (hunk 4 above)
Comment (Ormabe) [reply frontend/components/main/signup.js:71]: All styling was moved to a separate sign-up styling sheet

**Pull request merged**